### PR TITLE
Merge Request

### PR DESCRIPTION
**Pull request by the133448**

### CODE REVIEW

**Comment by the133448** [comment]
Wow, no idea how to find out what needs merging..

**Comment by nikooo777** [comment]
okay i see there several conficts here. any chances someone can checkout my master branch and see if we can make this a clean merge?

I suspect this PR doesn't have the color fixes yet some edits went into these files

**Comment by the133448** [comment]
Yep correct nikoo, I merged the color fixes beforehand

**Comment by 2called-chaos** [comment]
Yeah some of the conflicts are because he originally forked from another repository, kept merging features from this one and now pushes them back. Git can't figure out what's going on (due to diverged history) and it tries to reapply changes that are already applied. e.g. I saw my whole color changes in the conflicts

**Comment by M-archand** [comment]
@nikooo777 I tried out your pr66-merging branch and the colors/ranks work. The HUD box when you've left the start zone seems to work fine, speed doesn't go off center or disappear. However, the HUD when you're in the start zone is a bit bugged. This is what it looks like when you've left start zone: 
3j6prqh

But this is what it looks like when you haven't left the start zone: 
ctscthp

It looks like the elements aren't drawing properly, they just cycle over top of each other or something? I'm assuming it isn't meant to be this way, but if it is I don't think it looks very good in my personal opinion... but I feel like this must be a bug since the custom message at the bottom is doing some weird shit, can you confirm @the133448 ?

*Nvm, not a bug, it's supposed to be this way.

**Comment by 2called-chaos** [comment]
Yeah I can understand the appeal of these information cycling through but it does look very strange, especially because they scroll at different speeds.

**Comment by M-archand** [comment]
Not really a worry I guess, I just reverted the 2nd `else` statement on line 3147 to the original HUD configuration and it works fine.

**Comment by the133448** [comment]
Ok. gone through and fixed conflicts

**Comment by the133448** [comment]
@M-archand Ive made a cvar ck_custom_hud to toggle beetween the differnt displays in the hud 
b8b9c97aa29cd590933abe6117a27c792ca0d996 :) Ive also maded it so they all the data updates at once 08d530a0e369b488cce605bb9bcc92f89c0a139a

File: csgo/addons/sourcemod/scripting/ckSurf.sp [hunk 1]
```diff
@@ -30,9 +32,8 @@
 =           	 Definitions 		         =
 =============================================*/
 
-// Require new syntax and semicolons
+// Require new syntax
 #pragma newdecls required
-#pragma semicolon 1
```
Comment (nikooo777) [inline csgo/addons/sourcemod/scripting/ckSurf.sp]: why was this requirement removed? i think it's a step back

File: csgo/addons/sourcemod/scripting/ckSurf.sp [hunk 2]
```diff
@@ -812,7 +835,9 @@ public void OnLibraryRemoved(const char[] name)
 public void OnMapStart()
 {
 	hasStarted = true;
-	
+	char sBuffer[256];
+	GetConVarString(FindConVar("hostname"), sBuffer,sizeof(sBuffer));
+	Format(g_szServerNameBrowser, 128, sBuffer)
```
Comment (nikooo777) [inline csgo/addons/sourcemod/scripting/ckSurf.sp]: allowing things like this without semicolon is kinda bad practice

Comment (nikooo777) [reply csgo/addons/sourcemod/scripting/ckSurf.sp]: also i saw smlib was removed, seems to compile just fine when i take hooks.sp completely from your fork, any reasons? I guess it wasn't being used other than for a definition?

Comment (the133448) [reply csgo/addons/sourcemod/scripting/ckSurf.sp]: Im not sure, that would be something @Maxximou5 did.
For my changes past this point, should I make a new PR or just merge it to this one?

Comment (the133448) [reply csgo/addons/sourcemod/scripting/ckSurf.sp]: fixed in 1cf1d621966b71c2335bfa988995f5db0cd091da.

ask @Maxximou5  about why smlib was removed, He was the one who removed it

Comment (Maxximou5) [reply csgo/addons/sourcemod/scripting/ckSurf.sp]: ckSurf-misc.inc has everything we use that was from smlib. Out of the 350+ functions, we used 18. It's bad practice to use outdated libraries meant for other games when this is only for CS:GO. You could now use #pragma newdecls required on includes. There should be no reason to using old syntax either.

**Comment by the133448** [comment]
closed to move to #70

**Pull request closed without merging**